src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.cs

```diff
@@ -135,9 +139,25 @@ public CborReaderState Peek()
                 CborMajorType.Array => CborReaderState.StartArray,
                 CborMajorType.Map => CborReaderState.StartMap,
                 CborMajorType.Tag => CborReaderState.Tag,
-                CborMajorType.Special => CborReaderState.Special,
+                CborMajorType.Special => MapSpecialValueTagToReaderState(initialByte.AdditionalInfo),
                 _ => CborReaderState.FormatError,
             };
+
+            static CborReaderState MapSpecialValueTagToReaderState (CborAdditionalInfo value)
+            {
+                // https://tools.ietf.org/html/rfc7049#section-2.3
+
+                switch (value)
+                {
+                    case CborAdditionalInfo.SpecialValueNull: return CborReaderState.Null;
+                    case CborAdditionalInfo.SpecialValueFalse:
+                    case CborAdditionalInfo.SpecialValueTrue: return CborReaderState.Boolean;
```

Comment:
Newline and indent the return values, please.  Especially because of the multiple-label blocks.